src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorLanguageServer.cs

```diff
@@ -152,7 +153,10 @@ public static Task<ILanguageServer> CreateAsync(Stream input, Stream output, Tra
                         services.AddSingleton<RazorSemanticTokenInfoService, DefaultRazorSemanticTokenInfoService>();
                         services.AddSingleton<RazorHoverInfoService, DefaultRazorHoverInfoService>();
                         services.AddSingleton<HtmlFactsService, DefaultHtmlFactsService>();
-                    }));
+                    });
+                options.Serializer.Settings.Converters.Add(SemanticTokensOrSemanticTokensEditsConverter.Instance);
```

Comment:
The converter and `SemanticTokensOrSemanticTokensEdits` class are O#'s solution to the fact that this endpoint needs to be able to return 2 different classes. If the return value is just tokens it serializes that, if it's edits the Edits object gets serialized. I didn't put it into `RegisterRazorConverters` because that's in MS.ANC.Razor.LanguageServer.Common and it would mean I had to move all those models and the converter over into that Project, and that didn't seem like the correct place for it. I believe that I also tried adding it there and it didn't get used for whatever reason, but I can't swear to that because I was in the middle of a ton of edits.

Reply:
Oh no, I'm saying just move these lines up. it doesn't have to be in the RegisterJsonConverters method.